### PR TITLE
Add valset request by nonce support for QGB

### PR DESCRIPTION
**Pull request by SweeXordious**
Add valset request by nonce support for QGB.

Closes: https://github.com/celestiaorg/celestia-app/issues/343

### CODE REVIEW

**Comment by SweeXordious** [comment]
weird CI  not being triggered

File: go.mod [hunk 1]
```diff
@@ -18,7 +18,7 @@ require (
 	github.com/tendermint/tendermint v0.34.14
 	github.com/tendermint/tm-db v0.6.6
 	golang.org/x/net v0.0.0-20211015210444-4f30a5c0130f // indirect
-	golang.org/x/sys v0.0.0-20220412211240-33da011f77ad // indirect
+	golang.org/x/sys v0.0.0-20220422013727-9388b58f7150 // indirect
```
Comment (adlerjohn) [inline go.mod:21]: Is this a requirement for this PR? If not, it should be done in a separate PR.

Comment (SweeXordious) [reply go.mod]: done

**Review by evan-forbes — APPROVED** [review]
nice, I like this a lot :+1:
